src/main/java/eu/dissco/core/translator/terms/BaseDigitalObjectDirector.java

```diff
@@ -362,38 +359,32 @@ private String getOrganisationName(String organisationId) throws OrganisationExc
   private List<EntityRelationship> assembleDigitalSpecimenEntityRelationships(
       DigitalSpecimen ds) {
     var relationships = new ArrayList<EntityRelationship>();
-    relationships.add(getEntityRelationship(HAS_ORGANISATION_ID, ds.getOdsOrganisationID()));
-    relationships.add(getEntityRelationship(HAS_SOURCE_SYSTEM_ID, ds.getOdsSourceSystemID()));
-    relationships.add(getEntityRelationship(HAS_FDO_TYPE, fdoProperties.getDigitalSpecimenType()));
+    relationships.add(addEntityRelationship(HAS_ORGANISATION_ID, ds.getOdsOrganisationID(),
+        fdoProperties.getApplicationName(), fdoProperties.getApplicationPID()));
+    relationships.add(addEntityRelationship(HAS_SOURCE_SYSTEM_ID, ds.getOdsSourceSystemID(),
+        fdoProperties.getApplicationName(), fdoProperties.getApplicationPID()));
+    relationships.add(addEntityRelationship(HAS_FDO_TYPE, fdoProperties.getDigitalSpecimenType(),
+        fdoProperties.getApplicationName(), fdoProperties.getApplicationPID()));
     if (ds.getOdsPhysicalSpecimenIDType().equals(OdsPhysicalSpecimenIDType.RESOLVABLE)) {
       relationships.add(
-          getEntityRelationship(HAS_PHYSICAL_IDENTIFIER, ds.getOdsPhysicalSpecimenID()));
+          addEntityRelationship(HAS_PHYSICAL_IDENTIFIER, ds.getOdsPhysicalSpecimenID(),
+              fdoProperties.getApplicationName(), fdoProperties.getApplicationPID()));
     }
     if (ds.getDctermsLicense() != null && ds.getDctermsLicense().startsWith("http")) {
```

Comment:
hmm not sure. I would rather not have ERs that don't have a URI attached to it. So maybe we should have an chat if we should allow non-resolvable ERs at all and change the code accordingly. I will pick this up in the next exploration meeting. For now I would like to keep it this way. The distinction is the if the license doesn't start with http we won't make an ER at all, otherwise we would make an ER without a URI in it.